Serving input payload doc (#12848)
mlflow/mlflow: docs/source/model/signatures.rst

    Secondly, and just as importantly, it **validates the model's requirements**. This input example is utilized to execute a prediction 
    using the model that is about to be logged, thereby enhancing the accuracy in identifying model requirement dependencies.
    It is **highly recommended** to always include an input example along with your models when you log them.

>     Since MLflow 2.16.0, if the provided input example fails on validation, a warning is issued with an instruction to validate the input example

Reviewer: nit: not sure if we have to give a notice of "we will emit warning" here, as the warning itself is pretty self-explanatory...?

    prior to model deployment.

Since MLflow 2.16.0, when logging a model with an input example, there are two files saved into the model's artifacts directory:

- ``input_example.json``: The input example in JSON format.

> - ``serving_input_payload.json``: The input example converted to JSON format for querying a deployed model endpoint.

Reviewer: ```suggestion
- ``serving_input_payload.json``: The input example in JSON format, with additional transformation to have compatible schema for querying a deployed model REST endpoint.
```
nit: To make the difference even clearer...

The following example demonstrates the difference between the two files:

.. code-block:: python

    import mlflow

    class MyModel(mlflow.pyfunc.PythonModel):
        def predict(self, context, model_input, params=None):
            return model_input

    with mlflow.start_run():
        model_info = mlflow.pyfunc.log_model(
            python_model=MyModel(),
            artifact_path="model",
            input_example={"question": "What is MLflow?"},
        )

> Example files:

Reviewer: ```suggestion
Example files created by MLflow:
```

 +---------------------------------------+-----------------------------------------+
 |  input_example.json                   |  serving_input_payload.json             |

>  +---------------------------------------+-----------------------------------------+

Reviewer 2: Not a blocker, but I'd vertically stack the examples for editability

```
.. code-block::
   
     # input_example.json

.. code-block::
   
     # serving_input_payload.json
```

Reviewer: I think we can use [list-table](https://github.com/mlflow/mlflow/blob/d828b72897415548601e9825c43485e13136c96f/docs/source/llms/transformers/task.rst?plain=1#L77), much easier to edit imo

Author: Updated 
<img width="1343" alt="image" src="https://github.com/user-attachments/assets/8b4f3358-1687-4276-ba5c-d7997f4d2f4d">

 | .. code-block:: json                  | .. code-block:: json                    |
 |                                       |                                         |
 |    {                                  |   {                                     |
 |        "question": "What is MLflow?"  |       "inputs": {                       |
 |    }                                  |         "question": "What is MLflow?"   |
 |                                       |       }                                 |
 |                                       |   }                                     |
 |                                       |                                         |
 +---------------------------------------+-----------------------------------------+

> ``input_example.json`` contains the input example in its original format, while ``serving_input_payload.json`` is a JSON-serialized version of the input example with

Reviewer: nit: Maybe this description can go inside the table row.

> a defined ``key`` (one of ``dataframe_split, instances, inputs or dataframe_records``) that mlflow scoring server requires when `querying a deployed model endpoint <../../cli.html#mlflow-models-serve>`_.

Reviewer: ```suggestion
one of predefined keys (``"dataframe_split"``, ``"instances"``, ``"inputs"``, and ``"dataframe_records"``) that mlflow scoring server requires when `querying a deployed model endpoint <../../cli.html#mlflow-models-serve>`_.
```
super-nit

Reviewer: Btw the link for `cli.html` is broken. Can we link to https://mlflow.org/docs/latest/deployment/deploy-model-locally.html#local-inference-server-spec instead, which has more comprehensive guidance?

    It is **highly recommended** to always include an input example along with your models when you log them.

Since MLflow 2.16.0, when logging a model with an input example, there are two files saved into the model's artifacts directory:

- ``input_example.json``: The input example in JSON format.
- ``serving_input_payload.json``: The input example in JSON format, with additional transformation to have compatible schema for querying a deployed model REST endpoint.

The following example demonstrates the difference between the two files:

.. code-block:: python

    import mlflow

    class MyModel(mlflow.pyfunc.PythonModel):
        def predict(self, context, model_input, params=None):
            return model_input

    with mlflow.start_run():
        model_info = mlflow.pyfunc.log_model(
            python_model=MyModel(),
            artifact_path="model",
            input_example={"question": "What is MLflow?"},
        )

Example files logged by MLflow:

.. list-table::
   :header-rows: 1
   :class: wrap-table

   * - File name
     - Content
     - Explanation
   * - input_example.json
     - 
        .. code-block:: json

>             {
>                 "question": "What is MLflow?"
>             }

Reviewer 2: ```suggestion
            {
              "question": "What is MLflow?"
            }
```

     - The input example in its original format.
   * - serving_input_payload.json
     - 

>         .. code-block:: json
> 
>                 {
>                     "inputs": {
>                         "question": "What is MLflow?"
>                     }
>                 }

Reviewer 2: ```suggestion
        .. code-block:: json

            {
              "inputs": {
                "question": "What is MLflow?"
              }
            }
```
